test/Driver/EDMF/closures/entr_detr.jl

```diff
@@ -0,0 +1,56 @@
+#### Entrainment-Detrainment kernels
+
+function entr_detr(
+    ss::SingleStack{FT, N},
+    m::EntrainmentDetrainmentModel,
+    state::Vars,
+    diffusive::Vars,
+    aux::Vars,
+    t::Real,
+    direction,
+    i::Int
+) where {FT, N}
+
+    # Alias convention:
+    gm = state
+    en = state
+    up = state.edmf.updraft
+    gm_s = source
+    en_s = source
+    up_s = source.edmf.updraft
+
+    fill!(m.Λ, 0)
+    # precompute vars
+    ρinv = 1/gm.ρ
+    up_area = up[i].ρa/gm.ρ
+    w_up = up[i].u[3]
+    w_en = (gm.ρu[3]-sum([up[i].ρau[3] for i in 1:N]))*ρinv
+    b_up = up[i].buoyancy
+    b_en = (gm_a.buoyancy-sum([ρinv*up[i].ρa*up_a[i].buoyancy for i in 1:N]))
+    sqrt_tke = sqrt(max(en.tke,0.0))
```

Comment:
```suggestion
    sqrt_tke = sqrt(max(en.tke,0))
```

Reply:
(To not promote to `Float64`). This happens in a few places, so please adjust everywhere.